### DISCUSSION TITLE
Cloud Firstore Plugin Should Support Logical Or Query

### DISCUSSION THREAD

**Discussion by maheshj01** [firebase/flutterfire#4381] [Feature request]
Currently, we can do logical And operations using the where clause  something like this 
## Logical AND
```
Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1")
    .where("field2", isEqualTo: 'value2')
```
I would propose a similar functionality for logical OR 

## Proposal
```
Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1")
    .orWhere("field1", isEqualTo: 'value2')
```
The current approach which works for me is using the [CombineLatestStream](https://pub.dev/documentation/rxdart/latest/rx/CombineLatestStream-class.html) from the [rxdart ](https://pub.dev/packages/rxdart)package 

## Current Approach
```
final stream1 = Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value1");
final stream2 = Firestore.instance.collection("messages")
    .where("field1", isEqualTo: "value2");

StreamBuilder(
    stream: CombineLatestStream.list([
    stream0,
    stream1,
  ]),
builder: (context, snapshot) {
  final data0 = snapshot.data[0];
  final data1 = snapshot.data[1];
})
```
I am not sure how efficient the current approach is, or what are the technical challenges involved with adding support for Logical Or query to firestore plugin I would love to hear more about it. Feel free to close this issue if this is a duplicate or is not related.

**Comment by larssn** [comment]
You can already do that:

```
Firestore.instance.collection("messages")
    .where("field1", whereIn: ["value1", "value2"])
```

This is equivalent to your statement. It reads as "Find documents where field1's value is contained in the list of values (value1 and value2)".

You see some more examples here: https://firebase.googleblog.com/2019/11/cloud-firestore-now-supports-in-queries.html

Keep in mind that the Flutter plugin is limited to what the underlying Android/iOS/Web implementations support. It's a wrapper around those implementations.

**Reply by maheshj01** [reply]
Thanks

**Comment by maheshj01** [comment]
Update:

![image](https://user-images.githubusercontent.com/31410839/234727657-7eeee4d9-d6e0-4182-9892-6f56bdc07020.png)
Originally posted here https://stackoverflow.com/a/76018163
